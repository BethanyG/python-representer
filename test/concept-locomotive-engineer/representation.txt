Module([FunctionDef('placeholder_0', arguments([], [], arg('placeholder_1', lineno=4, col_offset=24, end_lineno=4, end_col_offset=28), [], [], defaults=[]), [Return(Call(Name('list', Load(), lineno=11, col_offset=11, end_lineno=11, end_col_offset=15), [Name('placeholder_1', Load(), lineno=11, col_offset=16, end_lineno=11, end_col_offset=20)], [], lineno=11, col_offset=11, end_lineno=11, end_col_offset=21), lineno=11, col_offset=4, end_lineno=11, end_col_offset=21)], [], lineno=4, col_offset=0, end_lineno=11, end_col_offset=21), FunctionDef('placeholder_2', arguments([], [arg('placeholder_3', lineno=14, col_offset=23, end_lineno=14, end_col_offset=37), arg('placeholder_4', lineno=14, col_offset=39, end_lineno=14, end_col_offset=53)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Tuple([Name('placeholder_5', Store(), lineno=22, col_offset=4, end_lineno=22, end_col_offset=9), Name('placeholder_6', Store(), lineno=22, col_offset=11, end_lineno=22, end_col_offset=17), Name('placeholder_7', Store(), lineno=22, col_offset=19, end_lineno=22, end_col_offset=29), Starred(Name('placeholder_8', Store(), lineno=22, col_offset=32, end_lineno=22, end_col_offset=36), Store(), lineno=22, col_offset=31, end_lineno=22, end_col_offset=36)], Store(), lineno=22, col_offset=4, end_lineno=22, end_col_offset=36)], Name('placeholder_3', Load(), lineno=22, col_offset=39, end_lineno=22, end_col_offset=53), lineno=22, col_offset=4, end_lineno=22, end_col_offset=53), Return(List([Name('placeholder_7', Load(), lineno=24, col_offset=12, end_lineno=24, end_col_offset=22), Starred(Name('placeholder_4', Load(), lineno=24, col_offset=25, end_lineno=24, end_col_offset=39), Load(), lineno=24, col_offset=24, end_lineno=24, end_col_offset=39), Starred(Name('placeholder_8', Load(), lineno=24, col_offset=42, end_lineno=24, end_col_offset=46), Load(), lineno=24, col_offset=41, end_lineno=24, end_col_offset=46), Name('placeholder_5', Load(), lineno=24, col_offset=48, end_lineno=24, end_col_offset=53), Name('placeholder_6', Load(), lineno=24, col_offset=55, end_lineno=24, end_col_offset=61)], Load(), lineno=24, col_offset=11, end_lineno=24, end_col_offset=62), lineno=24, col_offset=4, end_lineno=24, end_col_offset=62)], [], lineno=14, col_offset=0, end_lineno=24, end_col_offset=62), FunctionDef('placeholder_9', arguments([], [arg('placeholder_10', lineno=27, col_offset=22, end_lineno=27, end_col_offset=27)], kwonlyargs=[], kw_defaults=[], kwarg=arg('placeholder_11', lineno=27, col_offset=31, end_lineno=27, end_col_offset=37), defaults=[]), [Return(Dict([None, Constant('stops', lineno=35, col_offset=21, end_lineno=35, end_col_offset=28)], [Name('placeholder_10', Load(), lineno=35, col_offset=14, end_lineno=35, end_col_offset=19), Call(Name('list', Load(), lineno=35, col_offset=30, end_lineno=35, end_col_offset=34), [Call(Attribute(Name('placeholder_11', Load(), lineno=35, col_offset=35, end_lineno=35, end_col_offset=41), 'values', Load(), lineno=35, col_offset=35, end_lineno=35, end_col_offset=48), [], [], lineno=35, col_offset=35, end_lineno=35, end_col_offset=50)], [], lineno=35, col_offset=30, end_lineno=35, end_col_offset=51)], lineno=35, col_offset=11, end_lineno=35, end_col_offset=52), lineno=35, col_offset=4, end_lineno=35, end_col_offset=52)], [], lineno=27, col_offset=0, end_lineno=35, end_col_offset=52), FunctionDef('placeholder_12', arguments([], [arg('placeholder_10', lineno=38, col_offset=29, end_lineno=38, end_col_offset=34), arg('placeholder_13', lineno=38, col_offset=36, end_lineno=38, end_col_offset=58)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Dict([None, None], [Name('placeholder_10', Load(), lineno=46, col_offset=14, end_lineno=46, end_col_offset=19), Name('placeholder_13', Load(), lineno=46, col_offset=23, end_lineno=46, end_col_offset=45)], lineno=46, col_offset=11, end_lineno=46, end_col_offset=46), lineno=46, col_offset=4, end_lineno=46, end_col_offset=46)], [], lineno=38, col_offset=0, end_lineno=46, end_col_offset=46), FunctionDef('placeholder_14', arguments([], [arg('placeholder_15', lineno=49, col_offset=20, end_lineno=49, end_col_offset=31)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Tuple([List([Starred(Name('placeholder_16', Store(), lineno=56, col_offset=6, end_lineno=56, end_col_offset=13), Store(), lineno=56, col_offset=5, end_lineno=56, end_col_offset=13)], Store(), lineno=56, col_offset=4, end_lineno=56, end_col_offset=14), List([Starred(Name('placeholder_17', Store(), lineno=56, col_offset=18, end_lineno=56, end_col_offset=25), Store(), lineno=56, col_offset=17, end_lineno=56, end_col_offset=25)], Store(), lineno=56, col_offset=16, end_lineno=56, end_col_offset=26), List([Starred(Name('placeholder_18', Store(), lineno=56, col_offset=30, end_lineno=56, end_col_offset=39), Store(), lineno=56, col_offset=29, end_lineno=56, end_col_offset=39)], Store(), lineno=56, col_offset=28, end_lineno=56, end_col_offset=40)], Store(), lineno=56, col_offset=4, end_lineno=56, end_col_offset=40)], Call(Name('zip', Load(), lineno=56, col_offset=43, end_lineno=56, end_col_offset=46), [Starred(Name('placeholder_15', Load(), lineno=56, col_offset=48, end_lineno=56, end_col_offset=59), Load(), lineno=56, col_offset=47, end_lineno=56, end_col_offset=59)], [], lineno=56, col_offset=43, end_lineno=56, end_col_offset=60), lineno=56, col_offset=4, end_lineno=56, end_col_offset=60), Return(List([Name('placeholder_16', Load(), lineno=58, col_offset=12, end_lineno=58, end_col_offset=19), Name('placeholder_17', Load(), lineno=58, col_offset=21, end_lineno=58, end_col_offset=28), Name('placeholder_18', Load(), lineno=58, col_offset=30, end_lineno=58, end_col_offset=39)], Load(), lineno=58, col_offset=11, end_lineno=58, end_col_offset=40), lineno=58, col_offset=4, end_lineno=58, end_col_offset=40)], [], lineno=49, col_offset=0, end_lineno=58, end_col_offset=40)], [])