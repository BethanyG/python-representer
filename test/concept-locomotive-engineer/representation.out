# BEGIN TREE BEFORE
Module(
  [
    Expr(
      Constant(
        'Functions which helps the locomotive engineer to keep track of the train.',
        lineno=1,
        col_offset=0,
        end_lineno=1,
        end_col_offset=79),
      lineno=1,
      col_offset=0,
      end_lineno=1,
      end_col_offset=79),
    FunctionDef(
      'get_list_of_wagons',
      arguments(
        [],
        [],
        arg(
          'args',
          lineno=4,
          col_offset=24,
          end_lineno=4,
          end_col_offset=28),
        [],
        [],
        defaults=[]),
      [
        Expr(
          Constant(
            'Return a list of wagons.\n\n    :param *args: arbitrary number of wagons.\n    :return: list - list of wagons.\n    ',
            lineno=5,
            col_offset=4,
            end_lineno=9,
            end_col_offset=7),
          lineno=5,
          col_offset=4,
          end_lineno=9,
          end_col_offset=7),
        Return(
          Call(
            Name(
              'list',
              Load(),
              lineno=11,
              col_offset=11,
              end_lineno=11,
              end_col_offset=15),
            [
              Name(
                'args',
                Load(),
                lineno=11,
                col_offset=16,
                end_lineno=11,
                end_col_offset=20)],
            [],
            lineno=11,
            col_offset=11,
            end_lineno=11,
            end_col_offset=21),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=21)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=11,
      end_col_offset=21),
    FunctionDef(
      'fix_list_of_wagons',
      arguments(
        [],
        [
          arg(
            'each_wagons_id',
            lineno=14,
            col_offset=23,
            end_lineno=14,
            end_col_offset=37),
          arg(
            'missing_wagons',
            lineno=14,
            col_offset=39,
            end_lineno=14,
            end_col_offset=53)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Fix the list of wagons.\n\n    :param each_wagons_id: list - the list of wagons.\n    :param missing_wagons: list - the list of missing wagons.\n    :return: list - list of wagons.\n    ',
            lineno=15,
            col_offset=4,
            end_lineno=20,
            end_col_offset=7),
          lineno=15,
          col_offset=4,
          end_lineno=20,
          end_col_offset=7),
        Assign(
          [
            Tuple(
              [
                Name(
                  'first',
                  Store(),
                  lineno=22,
                  col_offset=4,
                  end_lineno=22,
                  end_col_offset=9),
                Name(
                  'second',
                  Store(),
                  lineno=22,
                  col_offset=11,
                  end_lineno=22,
                  end_col_offset=17),
                Name(
                  'locomotive',
                  Store(),
                  lineno=22,
                  col_offset=19,
                  end_lineno=22,
                  end_col_offset=29),
                Starred(
                  Name(
                    'rest',
                    Store(),
                    lineno=22,
                    col_offset=32,
                    end_lineno=22,
                    end_col_offset=36),
                  Store(),
                  lineno=22,
                  col_offset=31,
                  end_lineno=22,
                  end_col_offset=36)],
              Store(),
              lineno=22,
              col_offset=4,
              end_lineno=22,
              end_col_offset=36)],
          Name(
            'each_wagons_id',
            Load(),
            lineno=22,
            col_offset=39,
            end_lineno=22,
            end_col_offset=53),
          lineno=22,
          col_offset=4,
          end_lineno=22,
          end_col_offset=53),
        Return(
          List(
            [
              Name(
                'locomotive',
                Load(),
                lineno=24,
                col_offset=12,
                end_lineno=24,
                end_col_offset=22),
              Starred(
                Name(
                  'missing_wagons',
                  Load(),
                  lineno=24,
                  col_offset=25,
                  end_lineno=24,
                  end_col_offset=39),
                Load(),
                lineno=24,
                col_offset=24,
                end_lineno=24,
                end_col_offset=39),
              Starred(
                Name(
                  'rest',
                  Load(),
                  lineno=24,
                  col_offset=42,
                  end_lineno=24,
                  end_col_offset=46),
                Load(),
                lineno=24,
                col_offset=41,
                end_lineno=24,
                end_col_offset=46),
              Name(
                'first',
                Load(),
                lineno=24,
                col_offset=48,
                end_lineno=24,
                end_col_offset=53),
              Name(
                'second',
                Load(),
                lineno=24,
                col_offset=55,
                end_lineno=24,
                end_col_offset=61)],
            Load(),
            lineno=24,
            col_offset=11,
            end_lineno=24,
            end_col_offset=62),
          lineno=24,
          col_offset=4,
          end_lineno=24,
          end_col_offset=62)],
      [],
      lineno=14,
      col_offset=0,
      end_lineno=24,
      end_col_offset=62),
    FunctionDef(
      'add_missing_stops',
      arguments(
        [],
        [
          arg(
            'route',
            lineno=27,
            col_offset=22,
            end_lineno=27,
            end_col_offset=27)],
        kwonlyargs=[],
        kw_defaults=[],
        kwarg=arg(
          'kwargs',
          lineno=27,
          col_offset=31,
          end_lineno=27,
          end_col_offset=37),
        defaults=[]),
      [
        Expr(
          Constant(
            'Add missing stops to route dict.\n\n    :param route: dict - the dict of routing information.\n    :param **kwargs: arbitrary number of stops.\n    :return: dict - updated route dictionary.\n    ',
            lineno=28,
            col_offset=4,
            end_lineno=33,
            end_col_offset=7),
          lineno=28,
          col_offset=4,
          end_lineno=33,
          end_col_offset=7),
        Return(
          Dict(
            [
              None,
              Constant(
                'stops',
                lineno=35,
                col_offset=21,
                end_lineno=35,
                end_col_offset=28)],
            [
              Name(
                'route',
                Load(),
                lineno=35,
                col_offset=14,
                end_lineno=35,
                end_col_offset=19),
              Call(
                Name(
                  'list',
                  Load(),
                  lineno=35,
                  col_offset=30,
                  end_lineno=35,
                  end_col_offset=34),
                [
                  Call(
                    Attribute(
                      Name(
                        'kwargs',
                        Load(),
                        lineno=35,
                        col_offset=35,
                        end_lineno=35,
                        end_col_offset=41),
                      'values',
                      Load(),
                      lineno=35,
                      col_offset=35,
                      end_lineno=35,
                      end_col_offset=48),
                    [],
                    [],
                    lineno=35,
                    col_offset=35,
                    end_lineno=35,
                    end_col_offset=50)],
                [],
                lineno=35,
                col_offset=30,
                end_lineno=35,
                end_col_offset=51)],
            lineno=35,
            col_offset=11,
            end_lineno=35,
            end_col_offset=52),
          lineno=35,
          col_offset=4,
          end_lineno=35,
          end_col_offset=52)],
      [],
      lineno=27,
      col_offset=0,
      end_lineno=35,
      end_col_offset=52),
    FunctionDef(
      'extend_route_information',
      arguments(
        [],
        [
          arg(
            'route',
            lineno=38,
            col_offset=29,
            end_lineno=38,
            end_col_offset=34),
          arg(
            'more_route_information',
            lineno=38,
            col_offset=36,
            end_lineno=38,
            end_col_offset=58)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Extend route information with more_route_information.\n\n    :param route: dict - the route information.\n    :param more_route_information: dict - extra route information.\n    :return: dict - extended route information.\n    ',
            lineno=39,
            col_offset=4,
            end_lineno=44,
            end_col_offset=7),
          lineno=39,
          col_offset=4,
          end_lineno=44,
          end_col_offset=7),
        Return(
          Dict(
            [
              None,
              None],
            [
              Name(
                'route',
                Load(),
                lineno=46,
                col_offset=14,
                end_lineno=46,
                end_col_offset=19),
              Name(
                'more_route_information',
                Load(),
                lineno=46,
                col_offset=23,
                end_lineno=46,
                end_col_offset=45)],
            lineno=46,
            col_offset=11,
            end_lineno=46,
            end_col_offset=46),
          lineno=46,
          col_offset=4,
          end_lineno=46,
          end_col_offset=46)],
      [],
      lineno=38,
      col_offset=0,
      end_lineno=46,
      end_col_offset=46),
    FunctionDef(
      'fix_wagon_depot',
      arguments(
        [],
        [
          arg(
            'wagons_rows',
            lineno=49,
            col_offset=20,
            end_lineno=49,
            end_col_offset=31)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Fix the list of rows of wagons.\n\n    :param wagons_rows: list[tuple] - the list of rows of wagons.\n    :return: list[tuple] - list of rows of wagons.\n    ',
            lineno=50,
            col_offset=4,
            end_lineno=54,
            end_col_offset=7),
          lineno=50,
          col_offset=4,
          end_lineno=54,
          end_col_offset=7),
        Assign(
          [
            Tuple(
              [
                List(
                  [
                    Starred(
                      Name(
                        'row_one',
                        Store(),
                        lineno=56,
                        col_offset=6,
                        end_lineno=56,
                        end_col_offset=13),
                      Store(),
                      lineno=56,
                      col_offset=5,
                      end_lineno=56,
                      end_col_offset=13)],
                  Store(),
                  lineno=56,
                  col_offset=4,
                  end_lineno=56,
                  end_col_offset=14),
                List(
                  [
                    Starred(
                      Name(
                        'row_two',
                        Store(),
                        lineno=56,
                        col_offset=18,
                        end_lineno=56,
                        end_col_offset=25),
                      Store(),
                      lineno=56,
                      col_offset=17,
                      end_lineno=56,
                      end_col_offset=25)],
                  Store(),
                  lineno=56,
                  col_offset=16,
                  end_lineno=56,
                  end_col_offset=26),
                List(
                  [
                    Starred(
                      Name(
                        'row_three',
                        Store(),
                        lineno=56,
                        col_offset=30,
                        end_lineno=56,
                        end_col_offset=39),
                      Store(),
                      lineno=56,
                      col_offset=29,
                      end_lineno=56,
                      end_col_offset=39)],
                  Store(),
                  lineno=56,
                  col_offset=28,
                  end_lineno=56,
                  end_col_offset=40)],
              Store(),
              lineno=56,
              col_offset=4,
              end_lineno=56,
              end_col_offset=40)],
          Call(
            Name(
              'zip',
              Load(),
              lineno=56,
              col_offset=43,
              end_lineno=56,
              end_col_offset=46),
            [
              Starred(
                Name(
                  'wagons_rows',
                  Load(),
                  lineno=56,
                  col_offset=48,
                  end_lineno=56,
                  end_col_offset=59),
                Load(),
                lineno=56,
                col_offset=47,
                end_lineno=56,
                end_col_offset=59)],
            [],
            lineno=56,
            col_offset=43,
            end_lineno=56,
            end_col_offset=60),
          lineno=56,
          col_offset=4,
          end_lineno=56,
          end_col_offset=60),
        Return(
          List(
            [
              Name(
                'row_one',
                Load(),
                lineno=58,
                col_offset=12,
                end_lineno=58,
                end_col_offset=19),
              Name(
                'row_two',
                Load(),
                lineno=58,
                col_offset=21,
                end_lineno=58,
                end_col_offset=28),
              Name(
                'row_three',
                Load(),
                lineno=58,
                col_offset=30,
                end_lineno=58,
                end_col_offset=39)],
            Load(),
            lineno=58,
            col_offset=11,
            end_lineno=58,
            end_col_offset=40),
          lineno=58,
          col_offset=4,
          end_lineno=58,
          end_col_offset=40)],
      [],
      lineno=49,
      col_offset=0,
      end_lineno=58,
      end_col_offset=40)],
  [])

# BEGIN TREE AFTER
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [],
        arg(
          'placeholder_1',
          lineno=4,
          col_offset=24,
          end_lineno=4,
          end_col_offset=28),
        [],
        [],
        defaults=[]),
      [
        Return(
          Call(
            Name(
              'list',
              Load(),
              lineno=11,
              col_offset=11,
              end_lineno=11,
              end_col_offset=15),
            [
              Name(
                'placeholder_1',
                Load(),
                lineno=11,
                col_offset=16,
                end_lineno=11,
                end_col_offset=20)],
            [],
            lineno=11,
            col_offset=11,
            end_lineno=11,
            end_col_offset=21),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=21)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=11,
      end_col_offset=21),
    FunctionDef(
      'placeholder_2',
      arguments(
        [],
        [
          arg(
            'placeholder_3',
            lineno=14,
            col_offset=23,
            end_lineno=14,
            end_col_offset=37),
          arg(
            'placeholder_4',
            lineno=14,
            col_offset=39,
            end_lineno=14,
            end_col_offset=53)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Tuple(
              [
                Name(
                  'placeholder_5',
                  Store(),
                  lineno=22,
                  col_offset=4,
                  end_lineno=22,
                  end_col_offset=9),
                Name(
                  'placeholder_6',
                  Store(),
                  lineno=22,
                  col_offset=11,
                  end_lineno=22,
                  end_col_offset=17),
                Name(
                  'placeholder_7',
                  Store(),
                  lineno=22,
                  col_offset=19,
                  end_lineno=22,
                  end_col_offset=29),
                Starred(
                  Name(
                    'placeholder_8',
                    Store(),
                    lineno=22,
                    col_offset=32,
                    end_lineno=22,
                    end_col_offset=36),
                  Store(),
                  lineno=22,
                  col_offset=31,
                  end_lineno=22,
                  end_col_offset=36)],
              Store(),
              lineno=22,
              col_offset=4,
              end_lineno=22,
              end_col_offset=36)],
          Name(
            'placeholder_3',
            Load(),
            lineno=22,
            col_offset=39,
            end_lineno=22,
            end_col_offset=53),
          lineno=22,
          col_offset=4,
          end_lineno=22,
          end_col_offset=53),
        Return(
          List(
            [
              Name(
                'placeholder_7',
                Load(),
                lineno=24,
                col_offset=12,
                end_lineno=24,
                end_col_offset=22),
              Starred(
                Name(
                  'placeholder_4',
                  Load(),
                  lineno=24,
                  col_offset=25,
                  end_lineno=24,
                  end_col_offset=39),
                Load(),
                lineno=24,
                col_offset=24,
                end_lineno=24,
                end_col_offset=39),
              Starred(
                Name(
                  'placeholder_8',
                  Load(),
                  lineno=24,
                  col_offset=42,
                  end_lineno=24,
                  end_col_offset=46),
                Load(),
                lineno=24,
                col_offset=41,
                end_lineno=24,
                end_col_offset=46),
              Name(
                'placeholder_5',
                Load(),
                lineno=24,
                col_offset=48,
                end_lineno=24,
                end_col_offset=53),
              Name(
                'placeholder_6',
                Load(),
                lineno=24,
                col_offset=55,
                end_lineno=24,
                end_col_offset=61)],
            Load(),
            lineno=24,
            col_offset=11,
            end_lineno=24,
            end_col_offset=62),
          lineno=24,
          col_offset=4,
          end_lineno=24,
          end_col_offset=62)],
      [],
      lineno=14,
      col_offset=0,
      end_lineno=24,
      end_col_offset=62),
    FunctionDef(
      'placeholder_9',
      arguments(
        [],
        [
          arg(
            'placeholder_10',
            lineno=27,
            col_offset=22,
            end_lineno=27,
            end_col_offset=27)],
        kwonlyargs=[],
        kw_defaults=[],
        kwarg=arg(
          'placeholder_11',
          lineno=27,
          col_offset=31,
          end_lineno=27,
          end_col_offset=37),
        defaults=[]),
      [
        Return(
          Dict(
            [
              None,
              Constant(
                'stops',
                lineno=35,
                col_offset=21,
                end_lineno=35,
                end_col_offset=28)],
            [
              Name(
                'placeholder_10',
                Load(),
                lineno=35,
                col_offset=14,
                end_lineno=35,
                end_col_offset=19),
              Call(
                Name(
                  'list',
                  Load(),
                  lineno=35,
                  col_offset=30,
                  end_lineno=35,
                  end_col_offset=34),
                [
                  Call(
                    Attribute(
                      Name(
                        'placeholder_11',
                        Load(),
                        lineno=35,
                        col_offset=35,
                        end_lineno=35,
                        end_col_offset=41),
                      'values',
                      Load(),
                      lineno=35,
                      col_offset=35,
                      end_lineno=35,
                      end_col_offset=48),
                    [],
                    [],
                    lineno=35,
                    col_offset=35,
                    end_lineno=35,
                    end_col_offset=50)],
                [],
                lineno=35,
                col_offset=30,
                end_lineno=35,
                end_col_offset=51)],
            lineno=35,
            col_offset=11,
            end_lineno=35,
            end_col_offset=52),
          lineno=35,
          col_offset=4,
          end_lineno=35,
          end_col_offset=52)],
      [],
      lineno=27,
      col_offset=0,
      end_lineno=35,
      end_col_offset=52),
    FunctionDef(
      'placeholder_12',
      arguments(
        [],
        [
          arg(
            'placeholder_10',
            lineno=38,
            col_offset=29,
            end_lineno=38,
            end_col_offset=34),
          arg(
            'placeholder_13',
            lineno=38,
            col_offset=36,
            end_lineno=38,
            end_col_offset=58)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          Dict(
            [
              None,
              None],
            [
              Name(
                'placeholder_10',
                Load(),
                lineno=46,
                col_offset=14,
                end_lineno=46,
                end_col_offset=19),
              Name(
                'placeholder_13',
                Load(),
                lineno=46,
                col_offset=23,
                end_lineno=46,
                end_col_offset=45)],
            lineno=46,
            col_offset=11,
            end_lineno=46,
            end_col_offset=46),
          lineno=46,
          col_offset=4,
          end_lineno=46,
          end_col_offset=46)],
      [],
      lineno=38,
      col_offset=0,
      end_lineno=46,
      end_col_offset=46),
    FunctionDef(
      'placeholder_14',
      arguments(
        [],
        [
          arg(
            'placeholder_15',
            lineno=49,
            col_offset=20,
            end_lineno=49,
            end_col_offset=31)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Tuple(
              [
                List(
                  [
                    Starred(
                      Name(
                        'placeholder_16',
                        Store(),
                        lineno=56,
                        col_offset=6,
                        end_lineno=56,
                        end_col_offset=13),
                      Store(),
                      lineno=56,
                      col_offset=5,
                      end_lineno=56,
                      end_col_offset=13)],
                  Store(),
                  lineno=56,
                  col_offset=4,
                  end_lineno=56,
                  end_col_offset=14),
                List(
                  [
                    Starred(
                      Name(
                        'placeholder_17',
                        Store(),
                        lineno=56,
                        col_offset=18,
                        end_lineno=56,
                        end_col_offset=25),
                      Store(),
                      lineno=56,
                      col_offset=17,
                      end_lineno=56,
                      end_col_offset=25)],
                  Store(),
                  lineno=56,
                  col_offset=16,
                  end_lineno=56,
                  end_col_offset=26),
                List(
                  [
                    Starred(
                      Name(
                        'placeholder_18',
                        Store(),
                        lineno=56,
                        col_offset=30,
                        end_lineno=56,
                        end_col_offset=39),
                      Store(),
                      lineno=56,
                      col_offset=29,
                      end_lineno=56,
                      end_col_offset=39)],
                  Store(),
                  lineno=56,
                  col_offset=28,
                  end_lineno=56,
                  end_col_offset=40)],
              Store(),
              lineno=56,
              col_offset=4,
              end_lineno=56,
              end_col_offset=40)],
          Call(
            Name(
              'zip',
              Load(),
              lineno=56,
              col_offset=43,
              end_lineno=56,
              end_col_offset=46),
            [
              Starred(
                Name(
                  'placeholder_15',
                  Load(),
                  lineno=56,
                  col_offset=48,
                  end_lineno=56,
                  end_col_offset=59),
                Load(),
                lineno=56,
                col_offset=47,
                end_lineno=56,
                end_col_offset=59)],
            [],
            lineno=56,
            col_offset=43,
            end_lineno=56,
            end_col_offset=60),
          lineno=56,
          col_offset=4,
          end_lineno=56,
          end_col_offset=60),
        Return(
          List(
            [
              Name(
                'placeholder_16',
                Load(),
                lineno=58,
                col_offset=12,
                end_lineno=58,
                end_col_offset=19),
              Name(
                'placeholder_17',
                Load(),
                lineno=58,
                col_offset=21,
                end_lineno=58,
                end_col_offset=28),
              Name(
                'placeholder_18',
                Load(),
                lineno=58,
                col_offset=30,
                end_lineno=58,
                end_col_offset=39)],
            Load(),
            lineno=58,
            col_offset=11,
            end_lineno=58,
            end_col_offset=40),
          lineno=58,
          col_offset=4,
          end_lineno=58,
          end_col_offset=40)],
      [],
      lineno=49,
      col_offset=0,
      end_lineno=58,
      end_col_offset=40)],
  [])

# BEGIN NORMALIZED CODE AFTER
def placeholder_0(*placeholder_1):
    return list(placeholder_1)


def placeholder_2(placeholder_3, placeholder_4):
    placeholder_5, placeholder_6, placeholder_7, *placeholder_8 = placeholder_3
    return [placeholder_7, *placeholder_4, *placeholder_8, placeholder_5, placeholder_6]


def placeholder_9(placeholder_10, **placeholder_11):
    return {**placeholder_10, "stops": list(placeholder_11.values())}


def placeholder_12(placeholder_10, placeholder_13):
    return {**placeholder_10, **placeholder_13}


def placeholder_14(placeholder_15):
    [*placeholder_16], [*placeholder_17], [*placeholder_18] = zip(*placeholder_15)
    return [placeholder_16, placeholder_17, placeholder_18]
